### PR TITLE
Reconcile hackathon updates

### PR DESCRIPTION
**Pull request by tzerrell**
Since @carterbrett is not currently available to fix the merge conflicts in #10, I've created this PR to include that work, plus fix the merge conflicts. In places there was no reasonable way to merge without rewriting, so this adds some of my own writing as well. This also includes some minor fixes around links and names.

### CODE REVIEW

**Review by pdg744 — APPROVED** [review]
Great! Thanks for getting this over the line. I gave it a quick read and it looks good; will try it out as a user shortly.

File: README.md [hunk 1]
```diff
@@ -28,41 +35,48 @@ Get started writing your application by modifying these key files:
...
+### Test your zkVM program
+To check if your zkVM program will compile, and generate any errors if not, run the Rust compiler with
 
 ```bash
 cargo build
 ```
 
-While the following will build your Ethereum contracts.
-
+If you've written tests in your zkVM progam, run them with
```
Comment (pdg744) [inline README.md:47]: As a future improvement, it might be nice to include a sample test so that `cargo test` works out of the box.

Comment (pdg744) [reply README.md:47]: I guess it already works out of the box and just says 0 tests passed. /shrug

**Review by 3lkn — APPROVED** [review]
This looks great!

File: README.md [hunk 2]
```diff
@@ -28,41 +35,48 @@ Get started writing your application by modifying these key files:
...
+## Proving Modes
+The foundry template supports three different proving modes:
+1. `none` - By default, only the [executor](https://www.dev.risczero.com/docs/terminology#executor) runs your zkVM program and no proof is generated. Because there is no proving, this will be the fastest way to test. 
+2. `local` - A proof of execution generated on your local machine. 
+3. `bonsai` - A proof of execution is generated by the Bonsai API. 
 
-For offloading your [proof] requests to a local Bonsai instance, you can execute the tests as follows:
+Configure your preferred mode by setting the `BONSAI_PROVING` env variable.
+```bash
+export BONSAI_PROVING="local"
```
Comment (pdg744) [inline README.md:70]: seems odd that this line has quotes around `local` whereas the `bonsai` mode does not.

Comment (tzerrell) [reply README.md:70]: The effect will be equivalent but I agree this looks weird in the docs, I'll align to no-quotes here

File: README.md [hunk 3]
```diff
@@ -28,41 +35,48 @@ Get started writing your application by modifying these key files:
...
+```
```
Comment (pdg744) [inline README.md:71]: As a software noob this was a bit jarring for me. What do I do once I set the mode? 
```suggestion
``` 
After setting your preferred proving mode, use `cargo build`, `cargo test`, and `forge test` as explained above.
```

File: README.md [hunk 4]
```diff
@@ -28,41 +35,48 @@ Get started writing your application by modifying these key files:
...
```
Comment (pdg744) [inline README.md:72]: As a software noob, I would have appreciated a bit more hand-holding of what to do after setting the proving mode. 

```suggestion
After setting your preferred proving mode, use `forge test` to build and run your application. 
```

**Pull request merged**